<a href="https://colab.research.google.com/github/Deborah-NGOLO/projet-ventes-afrique-centrale/blob/main/01_data_cleaning_sales_marketing_central_Africa.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Analyse des ventes et performances marketing en Afrique centrale**

## Notebook 1 : Nettoyage et préparation des données

### Objectif
Ce notebook a pour objectif de :
- charger les données
- explorer la structure du dataset
- nettoyer les données
- renommer les colonnes
- convertir les types de données
- préparer une base propre pour l’analyse exploratoire

Le dataset contient des informations sur :
- les ventes
- les produits
- les clients
- les campagnes marketing
- les profits
- la satisfaction client

## Description des variables

| Variable | Description |
|---|---|
| date_commande | Date de la commande |
| pays | Pays où la vente a été réalisée |
| ville | Ville du client ou de la vente |
| canal_vente | Canal utilisé pour la vente (en ligne, magasin, distributeur, etc.) |
| segment_client | Catégorie de client |
| categorie_produit | Catégorie du produit vendu |
| produit | Nom du produit |
| quantite | Quantité vendue |
| prix_unitaire_fcfa | Prix unitaire du produit en FCFA |
| remise_pourcentage | Pourcentage de remise appliqué |
| chiffre_affaires_fcfa | Montant total des ventes en FCFA |
| cout_fcfa | Coût total associé à la vente |
| profit_fcfa | Profit généré par la vente |
| methode_paiement | Moyen de paiement utilisé |
| campagne_marketing | Campagne marketing associée |
| satisfaction_client | Niveau de satisfaction du client |

**1. importation des bibliothèques**

In [ ]:
import pandas as pd
import numpy as np
from google.colab import drive

In [ ]:
drive.mount('/content/drive')

df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/projet ventes marketing/sales_marketing_central_Africa.csv")

df

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


,order_id,date_commande,pays,ville,canal_vente,segment_client,categorie_produit,produit,quantite,prix_unitaire_fcfa,remise_pourcentage,chiffre_affaires_fcfa,cout_fcfa,profit_fcfa,methode_paiement,campagne_marketing,satisfaction_client
0,CMD-100001,2026-01-21,RCA,Bangui,Instagram,Particulier,Téléphones,iPhone 13,4,497017,15,1689858,976186,713672,Mobile Money,Promo rentrée,3.2
1,CMD-100002,2025-05-09,Cameroun,Yaoundé,Site web,PME,Alimentation,Riz 25kg,9,22793,15,174367,128919,45448,Mobile Money,Campagne Facebook Ads,3.4
2,CMD-100003,2025-02-17,Guinée équatoriale,Malabo,Facebook,Grande entreprise,Mode,Chemise homme,5,16899,0,84495,48804,35691,Espèces,Campagne Facebook Ads,3.2
3,CMD-100004,2025-01-10,Congo,Nkayi,Revendeur,Particulier,Alimentation,Riz 25kg,15,23871,10,322259,202751,119508,Mobile Money,Soldes Ramadan,4.2
4,CMD-100005,2025-01-17,Cameroun,Garoua,Facebook,Particulier,Téléphones,Tecno Spark 20,14,78646,0,1101044,740364,360680,Espèces,Soldes Ramadan,3.3
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
4995,CMD-104996,2026-04-22,Tchad,Sarh,Boutique physique,Grande entreprise,Alimentation,Lait en poudre,15,6418,5,91457,71544,19913,Virement bancaire,Aucune campagne,4.3
4996,CMD-104997,2024-10-09,Congo,Dolisie,WhatsApp Business,Particulier,Informatique,Dell Latitude,7,462332,15,2750876,2093090,657786,Virement bancaire,Promo WhatsApp,3.5
4997,CMD-104998,2024-04-21,Guinée équatoriale,Bata,WhatsApp Business,PME,Informatique,Dell Latitude,5,426964,0,2134820,1305979,828841,Virement bancaire,Aucune campagne,3.6
4998,CMD-104999,2024-11-04,RCA,Berbérati,Instagram,Grande entreprise,Informatique,Clé USB 64GB,7,9746,0,68222,39528,28694,Mobile Money,Promo WhatsApp,4.9


**2. Affichage des premières lignes**

In [ ]:
df.head()

,order_id,date_commande,pays,ville,canal_vente,segment_client,categorie_produit,produit,quantite,prix_unitaire_fcfa,remise_pourcentage,chiffre_affaires_fcfa,cout_fcfa,profit_fcfa,methode_paiement,campagne_marketing,satisfaction_client
0,CMD-100001,2026-01-21,RCA,Bangui,Instagram,Particulier,Téléphones,iPhone 13,4,497017,15,1689858,976186,713672,Mobile Money,Promo rentrée,3.2
1,CMD-100002,2025-05-09,Cameroun,Yaoundé,Site web,PME,Alimentation,Riz 25kg,9,22793,15,174367,128919,45448,Mobile Money,Campagne Facebook Ads,3.4
2,CMD-100003,2025-02-17,Guinée équatoriale,Malabo,Facebook,Grande entreprise,Mode,Chemise homme,5,16899,0,84495,48804,35691,Espèces,Campagne Facebook Ads,3.2
3,CMD-100004,2025-01-10,Congo,Nkayi,Revendeur,Particulier,Alimentation,Riz 25kg,15,23871,10,322259,202751,119508,Mobile Money,Soldes Ramadan,4.2
4,CMD-100005,2025-01-17,Cameroun,Garoua,Facebook,Particulier,Téléphones,Tecno Spark 20,14,78646,0,1101044,740364,360680,Espèces,Soldes Ramadan,3.3


**3. Affichage de la dimension du dataset**

In [ ]:
df.shape

(5000, 17)

Notre base de données contient 5000 observations reparties sur 17 colonnes.

**4. Information générales du dataset**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   order_id               5000 non-null   object 
 1   date_commande          5000 non-null   object 
 2   pays                   5000 non-null   object 
 3   ville                  5000 non-null   object 
 4   canal_vente            5000 non-null   object 
 5   segment_client         5000 non-null   object 
 6   categorie_produit      5000 non-null   object 
 7   produit                5000 non-null   object 
 8   quantite               5000 non-null   int64  
 9   prix_unitaire_fcfa     5000 non-null   int64  
 10  remise_pourcentage     5000 non-null   int64  
 11  chiffre_affaires_fcfa  5000 non-null   int64  
 12  cout_fcfa              5000 non-null   int64  
 13  profit_fcfa            5000 non-null   int64  
 14  methode_paiement       5000 non-null   object 
 15  camp

Le dataset est composé de variables de types variés : catégorielles, numériques et temporelles.

Les variables catégorielles telles que pays et ville permettront de segmenter les ventes afin d’identifier les zones géographiques les plus performantes.

Les variables quantitatives faciliteront les analyses statistiques (somme, moyenne, distribution des ventes), ce qui permettra d’évaluer les performances globales du business.

La variable date_commande est actuellement de type object. Elle devra être convertie au format datetime afin de permettre une analyse temporelle des ventes (évolution annuelle, mensuelle ou saisonnière).

Enfin, la variable order_id représente un identifiant unique de transaction. Elle ne contient pas d’information exploitable pour l’analyse statistique et sera donc exclue des analyses.

**5. Vérification des valeurs manquantes**

In [ ]:
df.isnull().sum()

,0
order_id,0
date_commande,0
pays,0
ville,0
canal_vente,0
segment_client,0
categorie_produit,0
produit,0
quantite,0
prix_unitaire_fcfa,0


Aucune valeur manquante n'a été recencé dans notre base de données

**6. Vérification des lignes dupliquées**

In [ ]:
df.duplicated().sum()

np.int64(0)

Aucune ligne dupliquée n’a été détectée dans le dataset.

**7. Statistiques descriptives**

In [ ]:
df.describe()

,quantite,prix_unitaire_fcfa,remise_pourcentage,chiffre_affaires_fcfa,cout_fcfa,profit_fcfa,satisfaction_client
count,5000.000000,5000.000000,5000.000000,5.000000e+03,5.000000e+03,5.000000e+03,5000.000000
mean,7.982000,135067.125200,5.055000,1.034843e+06,7.067983e+05,3.280451e+05,4.001800
std,4.334863,165290.433586,5.770497,1.571170e+06,1.080669e+06,5.205599e+05,0.584819
min,1.000000,3150.000000,0.000000,2.743000e+03,1.896000e+03,6.680000e+02,3.000000
25%,4.000000,20536.000000,0.000000,1.014650e+05,6.880625e+04,3.063225e+04,3.500000
50%,8.000000,52247.000000,5.000000,3.474435e+05,2.323250e+05,1.041285e+05,4.000000
75%,12.000000,157942.750000,10.000000,1.160222e+06,7.946202e+05,3.483720e+05,4.500000
max,15.000000,597897.000000,15.000000,8.966940e+06,6.889416e+06,3.492938e+06,5.000000


## Interprétation des statistiques descriptives

Les statistiques descriptives permettent de mieux comprendre la distribution des variables quantitatives du dataset.

### Observations :

- La quantité moyenne vendue est de **7.98**. Elle montre le volume moyen des commandes effectuées.
- Le chiffre d’affaires présente une forte dispersion, avec pour minimum **2.743000e+03 fcfa** et maximum **8.966940e+06 fcfa**.Cela indique des variations importantes entre les ventes.
- Certaines ventes génèrent des profits élevés tandis que d’autres sont moins rentables.
- Les remises appliquées varient selon les produits et les campagnes marketing.
- Les valeurs maximales observées suggèrent l’existence de ventes importantes ou de commandes exceptionnelles.

Ces premières observations permettent d’orienter les analyses exploratoires futures.

8. Suppression de la variable order_id

In [ ]:
df=df.drop('order_id',axis=1)

**9. Renommage de colonnes**

In [ ]:
df.columns = [
    "date_commande",
    "pays",
    "ville",
    "canal_vente",
    "segment_client",
    "categorie_produit",
    "produit",
    "quantite",
    "prix_unitaire_fcfa",
    "remise_pourcentage",
    "chiffre_affaires_fcfa",
    "cout_total_fcfa",
    "profit_net_fcfa",
    "methode_paiement",
    "campagne_marketing",
    "satisfaction_client"
]

**10. Vérification des noms de colonnes**

In [ ]:
df.head()

,date_commande,pays,ville,canal_vente,segment_client,categorie_produit,produit,quantite,prix_unitaire_fcfa,remise_pourcentage,chiffre_affaires_fcfa,cout_total_fcfa,profit_net_fcfa,methode_paiement,campagne_marketing,satisfaction_client
0,2026-01-21,RCA,Bangui,Instagram,Particulier,Téléphones,iPhone 13,4,497017,15,1689858,976186,713672,Mobile Money,Promo rentrée,3.2
1,2025-05-09,Cameroun,Yaoundé,Site web,PME,Alimentation,Riz 25kg,9,22793,15,174367,128919,45448,Mobile Money,Campagne Facebook Ads,3.4
2,2025-02-17,Guinée équatoriale,Malabo,Facebook,Grande entreprise,Mode,Chemise homme,5,16899,0,84495,48804,35691,Espèces,Campagne Facebook Ads,3.2
3,2025-01-10,Congo,Nkayi,Revendeur,Particulier,Alimentation,Riz 25kg,15,23871,10,322259,202751,119508,Mobile Money,Soldes Ramadan,4.2
4,2025-01-17,Cameroun,Garoua,Facebook,Particulier,Téléphones,Tecno Spark 20,14,78646,0,1101044,740364,360680,Espèces,Soldes Ramadan,3.3


**11. Conversion de la date**

In [ ]:
df["date_commande"] = pd.to_datetime(df["date_commande"])

12.Création des variables temporelles

In [ ]:
df["annee"] = df["date_commande"].dt.year
df["mois"] = df["date_commande"].dt.month

**13. Vérification final**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   date_commande          5000 non-null   datetime64[ns]
 1   pays                   5000 non-null   object        
 2   ville                  5000 non-null   object        
 3   canal_vente            5000 non-null   object        
 4   segment_client         5000 non-null   object        
 5   categorie_produit      5000 non-null   object        
 6   produit                5000 non-null   object        
 7   quantite               5000 non-null   int64         
 8   prix_unitaire_fcfa     5000 non-null   int64         
 9   remise_pourcentage     5000 non-null   int64         
 10  chiffre_affaires_fcfa  5000 non-null   int64         
 11  cout_total_fcfa        5000 non-null   int64         
 12  profit_net_fcfa        5000 non-null   int64         
 13  met

**13. Sauvegarde du dataset propre**

In [ ]:
df.to_csv("sales_marketing_central_africa_clean.csv", index=False)

In [ ]:
df.to_csv(
    "/content/drive/MyDrive/Colab Notebooks/projet ventes marketing/sales_marketing_central_africa_clean.csv",
    index=False
)

# Conclusion

Le dataset a été nettoyé et préparé pour l’analyse exploratoire.

Les principales opérations réalisées sont :
- la suppression des variables non pertinentes
- la conversion des types de données
- la création de variables temporelles
- la vérification de la qualité des données

Le dataset propre sera utilisé dans le prochain notebook pour réaliser l’analyse exploratoire des ventes et des performances marketing.